Lab3/app/src/main/java/com/example/battleship/GameStartActivity.kt

```diff
@@ -0,0 +1,503 @@
+package com.example.battleship
+
+import android.graphics.Color
+import android.os.Bundle
+import android.util.Log
+import android.view.View.INVISIBLE
+import android.view.View.VISIBLE
+import android.widget.Button
+import android.widget.TableLayout
+import android.widget.TableRow
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.button.MaterialButton
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+import com.google.firebase.database.ktx.getValue
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.StorageReference
+
+
+class GameStartActivity : AppCompatActivity() {
+    private var database: FirebaseDatabase? = null
+    lateinit var auth: FirebaseAuth
+    var gameID: Int = 0
+    var role: String? = ""
+    var antiRole: String? = ""
+    var storageRef: StorageReference? = null
+    lateinit var table: TableLayout
+    lateinit var enemy: TableLayout
+    lateinit var startBtn: Button
+    var ships: Int = 0
+    var cellsInt: Int = 0
+    var usedCells = Array(20) { i -> Array(2) { j -> 0 } }
+    var finalCells = Array(20) { i -> Array(2) { j -> 0 } }
```

Comment:
Game logic and DB operations should be in a ViewModel.

Reply:
move logic and DB's to ViewModel

Reply:
![image](https://user-images.githubusercontent.com/62778179/108451262-9025c080-7277-11eb-8c12-d6b4b53d0c1a.png)
![image](https://user-images.githubusercontent.com/62778179/108451298-9e73dc80-7277-11eb-94bf-1082bb38cb2e.png)